lib/evaluateFunctions.js

```diff
@@ -96,4 +99,389 @@ function absoluteValue(node) {
+    if (rootValue % exponentValue === 0) {
+      const newRootValue = rootValue/exponentValue;
+      const newRootNode = NodeCreator.constant(newRootValue);
+
+      newNode = NodeCreator.nthRoot(baseNode, newRootNode);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.CANCEL_EXPONENT, node, newNode);
+    }
+    else if (exponentValue % rootValue === 0) {
+      const newExponentValue = exponentValue/rootValue;
+      const newExponentNode = NodeCreator.constant(newExponentValue);
+
+      newNode = NodeCreator.operator('^', [baseNode, newExponentNode]);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.CANCEL_ROOT, node, newNode);
+    }
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Returns the nthRoot evaluated for a multiplication node.
+// Expects a multiplication node.
+function nthRootMultiplication(node) {
+  let newNode = clone(node, false);
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  let substeps = [];
+  let status;
+  if (NodeType.isConstant(rootNode) && !Negative.isNegative(rootNode)) {
+    status = factorChildren(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    status = groupTermsByRoot(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    status = convertMultiplicationToExponent(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+      if (newNode.args[0].op === '^') {
+        status = nthRootExponent(newNode);
+        substeps.push(status);
+        return NodeStatus.nodeChanged(
+          MathChangeTypes.NTH_ROOT_VALUE, node, status.newNode, true, substeps);
+      }
+    }
+  }
+
+  status = distributeNthRoot(newNode);
+  if (status.hasChanged()) {
+    substeps.push(status);
+    newNode = NodeStatus.resetChangeGroups(status.newNode);
+  }
+
+  status = evaluateNthRootForChildren(newNode);
+  if (status.hasChanged()) {
+    substeps.push(status);
+    newNode = NodeStatus.resetChangeGroups(status.newNode);
+
+    status = combineRoots(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.NTH_ROOT_VALUE, node, newNode, true, substeps);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given an nthRoot node with a constant positive root, will do the step of
+// factoring all the arguments under the radicand
+function factorChildren(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+
+  let children = [];
+  for (let i = 0; i < radicandNode.args.length; i++) {
+    const child = radicandNode.args[i];
+
+    if (NodeType.isConstant(child) && !Negative.isNegative(child)) {
```

Comment:
This wouldn't actually happen in the stepper as it would have been simplified, I'm only really testing for negative constants because we don't support complex numbers: sqrt(-1)